addons/main/functions/fnc_drawDownedUnitIndicator.sqf

```diff
@@ -8,16 +8,29 @@ private _color = [
-    if (_timeRemaining >= 0 && { GVAR(visibleBleedoutTimer) > 0 && { _isMedic || GVAR(visibleBleedoutTimer) == 2 } }) then {
-        drawIcon3D ["", GVAR(bleedoutTimerColor), (ASLtoAGL visiblePositionASL _x) vectorAdd [0, 0, 0.5], 0, 0, 0, str _timeRemaining, 2, 0.06 * (_size/2), "RobotoCondensed", "center", false, -0.0005 * (_size/2), -0.015 * (_size/2)];
-    };
-} forEach (GVAR(downedUnitIndicatorDrawCache) select {_x getVariable [QGVAR(unconscious), false]});
+if ((call CBA_fnc_currentUnit) getUnitTrait "Medic") then {
+    {
+        private _distance = _x distance _camPos;
+        if ( _distance > GVAR(showDownedUnitIndicatorRangeMedic) ) then {continue};
+        private _sizeFinal = linearConversion [0, GVAR(showDownedUnitIndicatorRangeMedic), _distance, _size, _size/10, false];
+        drawIcon3D [["\a3\ui_f\data\igui\cfg\actions\bandage_ca.paa", "\A3\ui_f\data\Map\VehicleIcons\pictureHeal_ca.paa"] select (_x getVariable [QGVAR(beingRevived), false]), _color, (ASLtoAGL visiblePositionASL _x) vectorAdd [0, 0, 0.5], _sizeFinal, _sizeFinal, 0, "", 0, 0, "RobotoCondensed", "center", true, 0, 0.5];
+       if (_distance > GVAR(bleedoutTimerRangeMedic) || {GVAR(visibleBleedoutTimer) < 1 }) then {continue};
+        private _timeRemaining = round ((_x getVariable [QGVAR(bleedoutKillTime), -1]) - cba_missionTime);
+        if (_timeRemaining >= 0) then {
+            drawIcon3D ["", GVAR(bleedoutTimerColor), (ASLtoAGL visiblePositionASL _x) vectorAdd [0, 0, 0.5], 0, 0, 0, str _timeRemaining, 2, 0.06 * (_size/2), "RobotoCondensed", "center", false, -0.0005 * (_size/2), -0.015 * (_size/2)];
+        };
+    } forEach (GVAR(downedUnitIndicatorDrawCache) select {_x getVariable [QGVAR(unconscious), false]});
+} else {
+    {
+        private _distance = _x distance _camPos;
+        if ( _distance > GVAR(showDownedUnitIndicatorRange) ) then {continue};
+        private _sizeFinal = linearConversion [0, GVAR(showDownedUnitIndicatorRange), _distance, _size, _size/10, false];
+        drawIcon3D [["\a3\ui_f\data\igui\cfg\actions\bandage_ca.paa", "\A3\ui_f\data\Map\VehicleIcons\pictureHeal_ca.paa"] select (_x getVariable [QGVAR(beingRevived), false]), _color, (ASLtoAGL visiblePositionASL _x) vectorAdd [0, 0, 0.5], _sizeFinal, _sizeFinal, 0, "", 0, 0, "RobotoCondensed", "center", true, 0, 0.5];
+       if (_distance > GVAR(bleedoutTimerRange) || {GVAR(visibleBleedoutTimer) < 2 }) then {continue};
+        private _timeRemaining = round ((_x getVariable [QGVAR(bleedoutKillTime), -1]) - cba_missionTime);
+        if (_timeRemaining >= 0) then {
+            drawIcon3D ["", GVAR(bleedoutTimerColor), (ASLtoAGL visiblePositionASL _x) vectorAdd [0, 0, 0.5], 0, 0, 0, str _timeRemaining, 2, 0.06 * (_size/2), "RobotoCondensed", "center", false, -0.0005 * (_size/2), -0.015 * (_size/2)];
+        };
+    } forEach (GVAR(downedUnitIndicatorDrawCache) select {_x getVariable [QGVAR(unconscious), false]});
+};
```

Comment:
Here I suggest instead of having an if else, which is sadly slower in sqf, you write the values that you need to local variables.

```suggestion
private _drawParams = [
    [GVAR(showDownedUnitIndicatorRange), GVAR(bleedoutTimerRange), 2], // rifleman values
    [GVAR(showDownedUnitIndicatorRangeMedic), GVAR(bleedoutTimerRangeMedic), 1] // medic values
] select ((call CBA_fnc_currentUnit) getUnitTrait "Medic");
_drawParams params ["_downedMaxRange", "_bleedoutMaxRange", "_visibleSetting"];

{
    private _distance = _x distance _camPos;
    if ( _distance > _downedMaxRange ) then {continue};
    private _sizeFinal = linearConversion [0, _downedMaxRange, _distance, _size, _size/10, false];
    drawIcon3D [["\a3\ui_f\data\igui\cfg\actions\bandage_ca.paa", "\A3\ui_f\data\Map\VehicleIcons\pictureHeal_ca.paa"] select (_x getVariable [QGVAR(beingRevived), false]), _color, (ASLtoAGL visiblePositionASL _x) vectorAdd [0, 0, 0.5], _sizeFinal, _sizeFinal, 0, "", 0, 0, "RobotoCondensed", "center", true, 0, 0.5];
    if (GVAR(visibleBleedoutTimer) > 0 && {_distance > _bleedoutMaxRange || {GVAR(visibleBleedoutTimer) < _visibleSetting }}) then {continue};
    private _timeRemaining = round ((_x getVariable [QGVAR(bleedoutKillTime), -1]) - cba_missionTime);
    if (_timeRemaining >= 0) then {
        drawIcon3D ["", GVAR(bleedoutTimerColor), (ASLtoAGL visiblePositionASL _x) vectorAdd [0, 0, 0.5], 0, 0, 0, str _timeRemaining, 2, 0.06 * (_size/2), "RobotoCondensed", "center", false, -0.0005 * (_size/2), -0.015 * (_size/2)];
    };
} forEach (GVAR(downedUnitIndicatorDrawCache) select {_x getVariable [QGVAR(unconscious), false]});
```